creator-node/src/snapbackSM/snapbackSM.js

```diff
@@ -1089,29 +1110,37 @@ class SnapbackSM {
      */
     this.syncDeDuplicator.removeSync(syncType, userWallet, secondaryEndpoint)
 
+    /**
+     * Do not issue syncRequest if SecondaryUserSyncFailureCountForToday already exceeded threshold
```

Comment:
i moved this logic from inside `additionalSyncIsRequired()` to here in `processSyncOperation()`

previously, it was inside additionalSyncIsRequired, but didn't do anything. bc - syncRequest is issued -> primary observes that max sync failure threshold is exceeded, but next snapback run will re-enqueue sync which will re-issue sync before checking additionalSyncIsRequired - therefore we still have infinite loop

now, `processSyncOp` will receive syncRequest job, but only issue syncRequest if threshold is not exceeded. each snapback interval will still infinitely re-enqueue syncRequest job, but it will never be issued

i think this is sufficient cc @dmanjunath @vicky-g

Reply:
nice catch!